PurchasesCoreSwiftTests/Networking/ETagManagerTests.swift

```diff
@@ -0,0 +1,232 @@
+import XCTest
+import Nimble
+import StoreKit
+
+import Purchases
+@testable import PurchasesCoreSwift
+
+class ETagManagerTests: XCTestCase {
+
+    private var mockUserDefaults: MockUserDefaults! = nil
+    private var eTagManager: ETagManager!
+    private var baseURL: URL = URL(string: "https://api.revenuecat.com")!
+    
+    override func setUp() {
+        super.setUp()
+        self.mockUserDefaults = MockUserDefaults()
+        self.eTagManager = ETagManager(userDefaults: self.mockUserDefaults)
+    }
+    
+    override func tearDown() {
+        self.mockUserDefaults = nil
+        self.eTagManager = nil
+    }
+    
+    func testETagIsEmptyIfThereIsNoETagSavedForThatRequest() {
+        let url = urlForTest()
+        let request = URLRequest(url: url)
+        let header = eTagManager.eTagHeader(for: request)
+        let value2: String? = header[ETagManager.eTagHeaderName]
+
+        expect(value2).toNot(beNil())
+        expect(value2) == ""
+    }
+
+    func testETagIsAddedToHeadersIfThereIsACachedETagForThatURL() {
+        let url = urlForTest()
+        let _ = mockStoredETagAndResponse(for: url)
+        let request = URLRequest(url: url)
+        let header = eTagManager.eTagHeader(for: request)
+        let value2: String? = header[ETagManager.eTagHeaderName]
+
+        expect(value2).toNot(beNil())
+        expect(value2) == "an_etag"
+    }
+
+    func testStoredResponseIsUsedIfResponseCodeIs304() {
+        let eTag = "an_etag"
+        let url = urlForTest()
+        let request = URLRequest(url: url)
+        let cachedResponse = mockStoredETagAndResponse(for: url, statusCode: HTTPStatusCodes.success, eTag: eTag)
+       
+        guard let httpURLResponse = HTTPURLResponse(url: url, statusCode: HTTPStatusCodes.notModifiedResponseCode.rawValue, httpVersion: "HTTP/1.1", headerFields: getHeaders(eTag: eTag)) else {
+            fatalError("Error initializing HTTPURLResponse")
+        }
```

Comment:
kinda surprised that the linter isn't complaining about line length here, do we have it disabled for test suites?

Reply:
I'd use line breaks per param to make this a bit easier to read though

Reply:
It is disabled for tests yes, there were too many changes and I disabled it for tests, we don't have detekt in Android tests either.